StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/UpgradeComponent.h

```diff
@@ -0,0 +1,69 @@
+#ifndef UPGRADECOMPONENT_H
+#define UPGRADECOMPONENT_H
+#include "ECS/Components/Component.h"
+#include <vector>
+#include <string>
+
+// Types of upgrades, for sorting/call purposes
+enum E_Type
+{
+	RANGE = 0,			// range weapon/attack upgrades
+	MELEE = 1,			// melee weapon/attack upgrades
+	MOVEMENT = 2,		// upgrades regarding movement.
+	STATS = 3,			// stat modifications such as hp change.
+	ONDAMAGE = 4,		// upgrades that are "triggered" when damage is taken.
+	ENEMYSPECIFIC = 5	// upgrades only relevant for enemy entities. example: enemy explode on death)
+};
```

Comment:
If an upgrade should be able to have multiple types, you should use flags instead of normal enums. Also the name : E_Type is not explaining what sort of enum it is. Type of what?